src/runtime.cpp

```diff
@@ -166,8 +171,16 @@ namespace hpx
         sigaction(SIGFPE, &new_action, nullptr);  // Floating point exception
         sigaction(SIGILL, &new_action, nullptr);  // Illegal instruction
         sigaction(SIGPIPE, &new_action, nullptr); // Bad pipe
-        sigaction(SIGSEGV, &new_action, nullptr); // Segmentation fault
+        #if __linux__ 
```

Comment:
Testing for __linux__ is not sufficient. As discussed on IRD, please see how this is done in other places, like: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/runtime_configuration.hpp#L139.